special-pages/pages/new-tab/app/components/BackgroundProvider.js

```diff
@@ -59,40 +60,56 @@ export function BackgroundConsumer({ browser }) {
     const { data } = useContext(CustomizerContext);
     const background = data.value.background;
 
-    switch (background.kind) {
-        case 'default': {
-            return <div className={styles.root} data-testid="BackgroundConsumer" data-background-kind="default" data-theme={browser} />;
+    useSignalEffect(() => {
+        const background = data.value.background;
+
+        // reflect some values onto the <body> tag
+        document.body.dataset.backgroundKind = background.kind;
+        let nextBodyBackground = '';
+
+        if (background.kind === 'gradient') {
```

Comment:
Just out of curiosity, was there a technical reason to not go with a `switch` or just personal preference?